tests/Http/ClientTest.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\Http;
+
+use Http\Client\Exception\NetworkException;
+use MeiliSearch\Client;
+use MeiliSearch\Exceptions\CommunicationException;
+use Psr\Http\Client\ClientInterface;
+use Psr\Http\Message\RequestInterface;
+use Tests\TestCase;
+
+class ClientTest extends TestCase
+{
+    public function testThrowCommunicationException(): void
+    {
+        $request = $this->getMockBuilder(RequestInterface::class)->getMock();
+        $networkException = new NetworkException('testThrowCommunicationException', $request);
+        $mockedHttpClient = $this->getMockBuilder(ClientInterface::class)->getMock();
```

Comment:
Thanks a lot for adding a new test that ensures the right error throw. May I ask why you chose to go with mocking?

Reply:
It was from inertia. Check the latest commit.

Reply:
btw, making the modifications I found out an issue :sweat_smile: (described in the commit message https://github.com/meilisearch/meilisearch-php/pull/144/commits/2b2469db4fbf2111ae3e6149cd641bd44f13d503)

Reply:
Nice!